crates/ruff/src/settings/types.rs

```diff
@@ -43,28 +44,6 @@ pub enum PythonVersion {
     Py311,
 }
 
-impl FromStr for PythonVersion {
-    type Err = anyhow::Error;
-
-    fn from_str(string: &str) -> Result<Self, Self::Err> {
-        match string {
-            "py33" | "py34" | "py35" | "py36" => {
-                warn_user_once!(
-                    "Specified a version below the minimum supported Python version. Defaulting \
-                     to Python 3.7."
-                );
-                Ok(Self::Py37)
-            }
```

Comment:
I think we had this custom implementation so that we could warn when `py33` and equivalent were provided, but IMO this is not a great tradeoff, since we lost (e.g.) the possible values in the CLI, at least out-of-the-box, by way of not using `ValueEnum`:

```diff
--- a/docs/configuration.md
+++ b/docs/configuration.md
@@ -211,7 +211,7 @@ Options:
       --format <FORMAT>
           Output serialization format for violations [env: RUFF_FORMAT=] [possible values: text, json, junit, grouped, github, gitlab, pylint, azure]
       --target-version <TARGET_VERSION>
-          The minimum Python version that should be supported
+          The minimum Python version that should be supported [possible values: py37, py38, py39, py310, py311]
```